Add error handling for user creation mutation

pedro-m-dietrich opened the pull request:
Error handling for invalid user creation, with their respective test cases, have been created.

pedro-m-dietrich committed:
Improve error handling for the createUser mutation

The error message format on invalid request to the `createUser` mutation
was improved by setting the error code (HTTP code) and adding a
`additionalInfo` field.

A specific message for an already registered e-mail has been added.

pedro-m-dietrich committed:
Add test cases for failed user creation mutation

Test cases for repeated e-mail, invalid password and invalid birth date
have been created.

alanraso commented on test/server.ts:
```diff
@@ -78,5 +87,57 @@ describe('Onboard server API', function () {
+      const currentUserCount = await prisma.user.count();
+
+      expect(currentUserCount).to.be.eq(previousUserCount);
+      expect(response.data.data.createUser).to.be.eq(null);
+      expect(response.data.errors[0].message).to.be.eq('E-mail is already in use.');
```
Sugeriria sempre tentar utilizar `to.be.deep.eq` no body completo da resposta, porque assim você testa se todos os campos do body estão de acordo com o que deveriam, e até se não faltaram ou ou se não tem campos a mais.

pedro-m-dietrich committed:
Use deep comparison in create user mutation tests

The test code has been refactored to use deep comparison with the API
responses in case of errors.

dan-ueno commented on test/server.ts:
```diff
@@ -78,5 +87,99 @@ describe('Onboard server API', function () {
+      const response = await axios.post(url, weakCreateUserMutation);
+      const currentUserCount = await prisma.user.count();
+
+      expect(currentUserCount).to.be.eq(previousUserCount);
```
acho que essas comparações poderiam ser simplificadas. Talvez não criar usuário algum no banco inicialmente e depois apenas conferir se continua sem nenhum usuário reduza as chamadas, além de garantir que não alterou um usuário existente, afinal se existe um count e ele está igual, mas !=0, o que garante que seu use-case não alterou algum já existente?

pedro-m-dietrich committed:
Reduce database calls in create user tests

Instead of comparing the number of entries in the database before and
after the acting, it is only compared to a hard-coded value.
